src/main/resources/pip-inspector.py

```diff
@@ -40,60 +43,71 @@
+                if not metadata_obj.url.endswith(".git"):
+                    continue
+                self.git_pkg_metadata[pkg] = metadata_obj.url
+
+    def resolve_git_package(self, requirement):
+        self._build_git_pkg_metadata()
+        for key in self.git_pkg_metadata:
+            if self.git_pkg_metadata[key] in requirement:
+                return resolve_package_by_name(key, [])
+        return None
 
-    for opt, arg in opts:
-        if opt in '--projectname':
-            project_name = arg
-        elif opt in '--requirements':
-            requirements_path = arg
+    def inspect(self):
+        if self.requirements_path is None:
+            print(self.project.render())
+            return
 
-    project = None
-
-    if project_name is not None:
-        project = resolve_package_by_name(project_name, [])
-
-    if project is None:
-        project = DependencyNode()
-        project.name = 'n?'
-        project.version = 'v?'
-
-    if requirements_path is not None:
-        try:
-            assert os.path.exists(requirements_path), ("The requirements file %s does not exist." % requirements_path)
-            requirements = parse_requirements(requirements_path, session=PipSession())
-            for req in requirements:
-                try:
-                    package_name = None
-                    # In 20.1 of pip, the requirements object changed
-                    if hasattr(req, 'req'):
-                        package_name = req.req.name
-                    if package_name is None:
-                        import re
-                        package_name = re.split('==|>=|<=|>|<', req.requirement)[0]
-
-                    requirement = resolve_package_by_name(package_name, [])
-                    if requirement is None:
-                        raise Exception()
-                    project.children = project.children + [requirement]
-                except:
-                    if req is not None and req.req is not None:
-                        print('--' + req.req.name)
-        except AssertionError:
-            print('r?' + requirements_path)
-        except:
-            print('p?' + requirements_path)
-
-    print(project.render())
+        requirements = parse_requirements(self.requirements_path, session=PipSession())
+        for req in requirements:
+            package_name = None
+            # In 20.1 of pip, the requirements object changed
+            if hasattr(req, "req"):
+                package_name = req.req.name
+            else:
+                package_name = req.requirement
+
+            if package_name.startswith("git+"):
+                requirement = self.resolve_git_package(package_name)
+            else:
+                pkg = re.split("==|>=|<=|>|<", req.requirement)[0]
+                requirement = resolve_package_by_name(pkg, [])
+
+            if requirement is None:
+                print("-- unknown requirement: %s" % package_name)
```

Comment:
I wasn't too sure about this line or if the `--` is interpreted as a comment when the output is parsed but it looked like something similar was happening here https://github.com/blackducksoftware/synopsys-detect/blob/master/src/main/resources/pip-inspector.py#L90 so I kept it but added more context to the output